src/components/Textarea/Textarea.module.scss

```diff
@@ -45,10 +45,19 @@
     }
   }
 
-  &--resize {
+  &__resize {
     resize: vertical;
   }
 
+  &__error {
+    box-shadow: none;
+    border: 1px solid $color-error;
+  }
+
+  &__label--error {
```

Comment:
tutaj np jest w porządku label to element podrzędny, a error to variant, można to zrobić jeszcze tak aby w przypadku kolejnych variantów do labela rozbudowywać je według struktury poniżej :P 
```

&__label {
   &--error {
     tutaj style
   }
}
```